webapi/src/main/scala/org/knora/webapi/routing/BaseEndpoints.scala

```
@@ -6,30 +6,92 @@
+      oneOfVariant[BadRequestException](statusCode(StatusCode.BadRequest).and(jsonBody[BadRequestException])),
+      oneOfVariant[ValidationException](statusCode(StatusCode.BadRequest).and(jsonBody[ValidationException])),
+      oneOfVariant[DuplicateValueException](statusCode(StatusCode.BadRequest).and(jsonBody[DuplicateValueException]))
+    )
+
+  private val secureDefaultErrorOutputs: EndpointOutput.OneOf[RequestRejectedException, RequestRejectedException] =
+    oneOf[RequestRejectedException](
+      // default
+      oneOfVariant[NotFoundException](statusCode(StatusCode.NotFound).and(jsonBody[NotFoundException])),
+      oneOfVariant[BadRequestException](statusCode(StatusCode.BadRequest).and(jsonBody[BadRequestException])),
+      oneOfVariant[ValidationException](statusCode(StatusCode.BadRequest).and(jsonBody[ValidationException])),
+      oneOfVariant[DuplicateValueException](statusCode(StatusCode.BadRequest).and(jsonBody[DuplicateValueException])),
+      // plus security
+      oneOfVariant[BadCredentialsException](statusCode(StatusCode.Unauthorized).and(jsonBody[BadCredentialsException])),
+      oneOfVariant[ForbiddenException](statusCode(StatusCode.Forbidden).and(jsonBody[ForbiddenException]))
     )
 
   val publicEndpoint = endpoint.errorOut(defaultErrorOutputs)
+
+  val securedEndpoint = endpoint
+    .errorOut(secureDefaultErrorOutputs)
+    .securityIn(auth.bearer[Option[String]](WWWAuthenticateChallenge.bearer))
+    .securityIn(cookie[Option[String]](authenticator.calculateCookieName()))
+    .securityIn(auth.basic[Option[UsernamePassword]](WWWAuthenticateChallenge.basic("realm")))
+    .serverSecurityLogic {
+      case (Some(jwtToken), _, _) => authenticateJwt(jwtToken)
+      case (_, Some(cookie), _)   => authenticateJwt(cookie)
+      case (_, _, Some(basic))    => authenticateBasic(basic)
+      case _                      => Future.successful(Left(BadCredentialsException("No credentials provided.")))
+    }
```

are you silently dropping support for authenticating with query parameters? If so, I applaud you :D

---

Yes, this is not implemented for the admin/project endpoints. 
I would like to drop support for that in the long run entirely.
I guess we will have to figure out if it actually still in use.
Should be worth noting on the release notes.

---

also, docs need to be updated

---

Good point, updated the documentation in 9ed8a02d052682f3a320ad24f22c73bf7610ec43